lib/src/array.rs

```diff
@@ -23,7 +25,41 @@ pub(crate) trait YrsArrayObservationDelegate: Send + Sync + Debug {
     fn call(&self, value: Vec<YrsChange>);
 }
 
+// unsafe impl Send for YrsArrayIterator {}
```

Comment:
What's the thinking re: the iterator? Are we considering exposing it as a first-class instance through the rust language bindings down the road? (That might make some of the other pieces Swift conformance pieces a bit easier, as they build on top of iterators).

Reply:
Yes, it was my initial plan to expose it as a first-class instance to avoid the pre-computing array using `each` and then iterating through the pre-computed array, because I believe it is far from perfect. Unfortunately, I couldn't make it work in the time I allocated myself to do it.  

So I decided to leave it for the next release. But I believe it definitely needs to be done as it will make conforming to Swift collection protocols much easier.